ocaml/xapi/workload_balancing.ml

```diff
@@ -114,6 +115,66 @@ let assert_wlb_enabled ~__context =
   if not (Db.Pool.get_wlb_enabled ~__context ~self:pool) then
     raise_disabled ()
 
+let read_file path =
+  let ic = open_in path in
+  Fun.protect
+    ~finally:(fun () -> close_in ic ; Sys.remove path)
+    (fun () ->
+      let length = in_channel_length ic in
+      really_input_string ic length
+    )
+
+let fetch_certificate wlb_url =
+  Ssl.init () ;
```

Comment:
You also need to call `Ssl_threads.init ()` globally, or pass `thread_safe:true` here.
By default the ssl module (and OpenSSL) is *not* thread-safe as a performance optimization.
But XAPI is a multithreaded application so we must always use this in a thread-safe way